### ISSUE TITLE
Actionbars Overhaul

### ISSUE THREAD

**Issue by kaytotes** [kaytotes/ImprovedBlizzardUI#49]
In Battle for Azeroth the Blizzard UI is getting a few tweaks to the action bars to honestly look much more like this addon; squishing the action bars, moving the bags and micro menu (though still displayed on screen) and placing the experience bar beneath the buttons (whereas I have it on top).

Frankly since Blizzard is going to be implementing most of this themselves once the expansion comes out I will essentially gut this again to primarily use their code. In the mean time however I really like how Blizzard have done it in general so I want to recreate it now and gut later.

**Battle for Azeroth Screenshot**
![vha3pkz](https://user-images.githubusercontent.com/7526918/35247249-9a192a4a-ffc1-11e7-87cf-da388ef20cbc.jpg)

**Comment by kaytotes** [comment]
Started working on this earlier today and I'm pretty confident I've managed to replicate the BfA UI. For now Artifact Power displays where the exp bar is when on a character with an artifact but other besides that both the Honor Bar and Reputation Bars have been disabled as I can't find any details on how Blizzard are handling these with their UI in the next expansion. Once I can get more details on that I'll tweak this to match.

![screenshot_1](https://user-images.githubusercontent.com/7526918/35477741-7c991e1c-03c2-11e8-9f92-6be1449f6925.png)

Includes Out of Range Indication
![screenshot_2](https://user-images.githubusercontent.com/7526918/35477743-80d59078-03c2-11e8-9f1c-b7037c34ecb4.png)

Stance / Forms Bar Positioning
![screenshot_3](https://user-images.githubusercontent.com/7526918/35477744-84a16056-03c2-11e8-865e-64772466c5d1.png)

Pet Bar Positioning
![screenshot_4](https://user-images.githubusercontent.com/7526918/35477745-87de5f30-03c2-11e8-8aae-b26b4247d2c9.png)

Also included a toggle for displaying Macro / Hotkey text on a per action bar basis.
![screenshot_5](https://user-images.githubusercontent.com/7526918/35477746-8b58bb06-03c2-11e8-95c8-015f71b31837.png)

By default the action bars are now smaller than they were before but i've also included a scale config for those that want it back to the larger style.
![screenshot_6](https://user-images.githubusercontent.com/7526918/35477748-8e766f04-03c2-11e8-9789-4e437b40c931.png)

Fixed Pet Bar Micro Menu. Really I want to change the art to properly accommodate the UI but for now at least for Pet Battlers its a bit easier.
![screenshot_7](https://user-images.githubusercontent.com/7526918/35478036-b4fae030-03ca-11e8-8da6-14dc1b329e45.png)

Known Issues: No Micro Menu for Vehicle UI. Should either make this appear or modify the art to match the rest of the UI properly.

**Comment by kaytotes** [comment]
Been busy with work but doing some more tweaking this weekend. Been watching BfA alpha footage and the new action bar is pretty damned close to blizzards. Might do some pixel counting to get exact scales and numbers. 

Pet bar seems to be in a different position to mine however.

![screenshot_1](https://user-images.githubusercontent.com/7526918/36001552-84726476-0d1e-11e8-8a53-72a6707be78f.png)

As is the stance bar.

![image](https://user-images.githubusercontent.com/7526918/36001637-d9242eaa-0d1e-11e8-9cd5-7ca3c61588b3.png)

Seems that when the "Bottom Right" action bar is shown the UI expands to be bigger (as above) but I sincerely don't like that so I'm not going to be implementing that and will be maintaining the central stacking.

Need to find footage of a class that has both a stance bar and a pet bar to see how Blizzard handles this. **Can anyone find footage of this? I don't play any of those classes so maybe Shadow Priest, DK?** It's a small use case but needs covering nonetheless.

**Comment by sphvn** [comment]
DK no longer has stances / presences, will try pull off the Shadow Priest one this evening.

**Comment by sphvn** [comment]
Shadow Priest only have the stance bar for shadow form, they don't seem to get a pet bar for shadow fiend or mind bender.

**Comment by kaytotes** [comment]
@sphvn That confirms what I managed to piece together from YouTube too. The last time I played a shadow priest was in early WoD which was when I even learned that a class could have both stance and pet bar. Hell this was the commit 2 years ago https://github.com/kaytotes/ImprovedBlizzardUI/commit/97fdee072aa0d8980dd2f0b66917f4b5a4e48aeb

I think what I'll have to do is stick with the current positioning of the Stance Bar (Left) and Pet Bar (Right) to avoid overlap currently in Legion and then in BfA adjust for the class changes and position things properly. If I end up somehow getting BfA access then i'll make a seperate branch for it.
